tests/components/heos/test_config_flow.py

```diff
@@ -47,27 +47,29 @@
 
 async def test_create_entry_when_host_valid(hass, controller):
     """Test result type is create entry when host is valid."""
-    flow = HeosFlowHandler()
-    flow.hass = hass
     data = {CONF_HOST: "127.0.0.1"}
-    result = await flow.async_step_user(data)
+    result = await hass.config_entries.flow.async_init(
+        heos.DOMAIN, context={"source": "user"}, data=data
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
     assert result["title"] == "Controller (127.0.0.1)"
     assert result["data"] == data
+    await heos.async_setup_entry(hass, result)
```

Comment:
We shouldn't call this. We should patch it. See eg the hue config flow tests.

https://github.com/home-assistant/home-assistant/blob/638a3025df1ef85a4cb53244a223d9d629381911/tests/components/hue/test_config_flow.py#L19-L23